amazon-s3/5.6/modules/ROOT/pages/amazon-s3-connector-config-topics.adoc

```diff
@@ -9,6 +9,151 @@ In the `Create Object` operation, set the `Content Length` to a value greater th
 
 To encrypt objects that you want to store in S3 buckets using customer managed master keys, specify a Customer Master Key ID in the `KMS Master Key` field in the `Create Object` configuration.
 
+== Prerequisites for Source Operations
+
+There are two scenarios for the `On New Object` and `On Deleted Object` source operations:
+
+. The <<queue-is-created,queue is created>> and specified
+. The <<queue-created-by-source,queue will be created>> by the source
+
+[[queue-is-created]]
+== Queue is Already Created
+
+To correctly use a source operation that already exists, you must enable these actions for the Amazon SQS queue:
+
+[source,json,linenums]
+----
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "VisualEditor0",
+            "Effect": "Allow",
+            "Action": [
+                "sqs:DeleteMessage",
+                "sqs:GetQueueUrl",
+                "sqs:ReceiveMessage",
+                "sqs:GetQueueAttributes"
+            ],
+            "Resource": "arn:aws:sqs:REGION:ACCOUNT_ID:QUEUE_NAME"
+        }
+    ]
+}
+----
+
+Additionally, to enable the queue to receive messages when a new object is created in an S3 bucket, you must create permission for the action in this format:
+
+[source,json,linenums]
+----
+{
+  "Version": "2012-10-17",
+  "Id": "arn:aws:sqs:REGION:ACCOUNT_ID:QUEUE_NAME/SQSDefaultPolicy",
+  "Statement": [
+    {
+      "Sid": "Sid1593770403887",
+      "Effect": "Allow",
+      "Principal": "*",
+      "Action": "SQS:SendMessage",
+      "Resource": "arn:aws:sqs:REGION:ACCOUNT_ID:QUEUE_NAME",
+      "Condition": {
+        "ArnLike": {
+          "aws:SourceArn": "arn:aws:s3:::BUCKET_NAME"
+        }
+      }
+    }
+  ]
+}
+----
+
+[[queue-created-by-source]]
+== Queue Created by Source Operation
+
+If the queue is created by the source operation, the queue is created with the prefix `MULE-S3-TRIGGER`. 
+
+In this scenario, you must have permission to create a queue and to also set the queue attributes:
```

Comment:
```suggestion
Creating the queue by using a source operation requires permission to both create the queue and set the queue attributes:
```

Reply:
Because Aws works that way. There is difference between user permission and queue policy. When admin on aws creates a user by default he does not have any permission at all. So we need to give him some kind of permission. In this case we need to give him permissions (the minimum) to create queue and other actions in order to invoke operations. 

example: operation provided by SQS - create queue - user needs permission with action **create queue**
source operation provided by S3 - behind the scene we are using create queue, delete message operations etc. Therefore, user who is using connector does not know what type of permission he needs for source operation